core/src/main/java/io/questdb/griffin/engine/functions/GroupByFunction.java

```diff
@@ -76,6 +82,10 @@ default boolean earlyExit(MapValue mapValue) {
         return false;
     }
 
+    default int getSampleByFlags() {
+        return SAMPLE_BY_FILL_VALUE | SAMPLE_BY_FILL_NONE | SAMPLE_BY_FILL_NULL | SAMPLE_BY_FILL_PREVIOUS;
```

Comment:
What's going on in my head: the default impl does not provide some fill types. e.g. `SAMPLE_BY_FILL_LINEAR`. 
This tells me there are requirements a group by function must fulfil to work with linear. What are these requirements? I can see impls like `ApproxCountDistinctIPv4GroupByFunction` override this method to return `SAMPLE_BY_FILL_ALL`. What makes these impls special? 

In other words: As a group-by function implementer, I'd like to see some guidance on when I can return what.

Reply:
the requirements will obvious as soon as function is tested with linear fill. This fill calls various "set" methods, depending on type of the argument, that by default will produce "unsupported operation exception"